### ISSUE TITLE
Document how stuff implementing DataSerializable serializes to config for

### ISSUE THREAD

**Issue by Katrix-** [SpongePowered/SpongeDocs#488]
Making an issue here so that discussion on this can be done here instead on the forums. 

Link with idea: https://forums.spongepowered.org/t/lets-get-chat-formatting-right/12613/17?u=katrix 

Copy-Paste from thread: 
The second thing I think we need to think about is a bit documentation. I'm not talking about plugin documentation or developer documentation. What I've more referring to here is documentation of things like TextTemplate and Text that players will see again and again in their configs. I don't know how far we should go with this, but I feel like we need some central place where players can understand how Text and TextTemplate works, not just from a development standpoint, but also from an user standpoint. 

So basically, we need to look at how we can help users not feel alienated and confused when they open up the config and see lots of stuff that they have no idea what does. I think the docs would be a nice place for this as it should be consistent between plugins.

**Comment by Inscrutable** [comment]
This could use a more meaningful title, and some context from the thread. It might get a page in the docs _if_ I can just figure out *what* is being requested, and where. Paging @ryantheleach for some assistance...

**Comment by Katrix-** [comment]
I'm not good with titles or names :(

The main idea here is to provide documentation for users that does not rely on the assumption that the person reading the docs is a developer, but instead an user wanting to run a Sponge server. 

Just because they are are user doesn't mean that they won't run in `Text`, `TextTexmplate` and other `DataSerializable`. Instead they will find these most of the time in their configs. So, the idea goes that if normal users will find these in their configs for example, there should be documentation that they can read up on to find out how to manipulate these.

**Comment by ryantheleach** [comment]
I might look at this but first I need to fix my workspace, test kashikes annotated commands. fix the config docs, finish the ray tracing docs.

**Comment by RobertHerhold** [comment]
@ryantheleach What needs done with Ray tracing? I thought I included everything

**Comment by ryantheleach** [comment]
Yo might have fixed it up since i last looked.

**Comment by gabizou** [comment]
I think the issue should be renamed so as to not be confused with `DataSerializable`s versus `ConfigSerializable`s (or registered through `TypeSerializer`s).
